src/config/context/model.ts

```diff
@@ -0,0 +1,58 @@
+import { injectable, registry, } from 'tsyringe';
+import fs from 'fs';
+import { CONTEXTFILE_PATH } from '../../constants';
+
+export interface IContext {
+  readonly current?: string,
+  readonly store: {
+    [name: string]: string
+  }
+}
+
+export class Context implements IContext {
+  current?: string | undefined;
+  store: { [name: string]: string; };
+
+  constructor(ctx: IContext) {
+    this.current = ctx.current;
+    this.store = ctx.store;
+  }
+
+  getContext(contextName: string) {
+    return this.store[contextName as string];
+  }
+}
+
+export interface IContextAllocator {
+  contextFilePath?: string
+  load: () => Context | undefined
+  save: (context: Context) => boolean
+}
+
+@injectable()
+@registry([
+  {
+    token: 'IContextAllocator',
+    useToken: ContextAllocator
+  }
+])
+export class ContextAllocator implements IContextAllocator {
+  contextFilePath = CONTEXTFILE_PATH;
+  load(): Context | undefined {
+    try {
+      return new Context(JSON.parse(fs.readFileSync(this.contextFilePath, 'utf8')) as IContext);
+    } catch (error) {
+      return undefined;
+    }
+  }
+
+  save(context: Context) {
+    try {
+      fs.writeFileSync(this.contextFilePath, JSON.stringify(context), { encoding: 'utf8' });
```

Comment:
I prefer to explicit stringify only the properties from context `current` and `store`, not whole `context` object. It's safer. Why? Because if someone will write the function in class with arrow notation, then it will be treated as reference to the function, not method in the prototype of class. See example:

![image](https://user-images.githubusercontent.com/20404945/133992679-c2a6a532-3b28-4651-b730-8e8498feb34e.png)

As you can see, arrow function is treated as normal property, not as method, so please change code to the:

```suggestion
      fs.writeFileSync(this.contextFilePath, JSON.stringify({ current: context.current, store: context.store }), { encoding: 'utf8' });
```